src/main/java/com/commercetools/service/paypalPlus/impl/PaypalPlusPaymentServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.commercetools.service.paypalPlus.impl;
+
+import com.commercetools.exception.PaypalPlusServiceException;
+import com.commercetools.service.paypalPlus.PaypalPlusPaymentService;
+import com.paypal.api.payments.Payment;
+import com.paypal.base.rest.APIContext;
+import com.paypal.base.rest.PayPalRESTException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * <b>Note:</b> since {@link APIContext} caches <i>PayPal-Request-Id</i> one should not reuse the same instance of
+ * {@link PaypalPlusPaymentServiceImpl} for multiple Paypal requests.
+ */
+@Service
+@Scope("prototype")
+public class PaypalPlusPaymentServiceImpl extends BasePaypalPlusService implements PaypalPlusPaymentService {
```

Comment:
good application design is  when you don't need to change an old class, when you have to implement a new one. Basically, _Payment_ and potential future _Refund_/_Notify_/_Whatever_ classes are independent, so we don't want to touch _Payment_ implementation when a new independent class is needed.

Maybe you are right and we won't have new classes, based on _ BasePaypalPlusService_, but i expect the extension soon and i'd like to found some flexibility for future developers.